tests/frontend/test_io.py

```diff
@@ -335,6 +337,27 @@ def test_free_par_str(self):
         assert bb.operations[0] == {"op": "Sgate", "modes": [0], "args": ['{r}', 0.0], "kwargs": {}}
         assert bb.operations[1] == {"op": "Zgate", "modes": [1], "args": ['3*log(-{alpha})'], "kwargs": {}}
 
+    def test_tdm_program(self):
+        prog = TDMProgram(2)
+
+        with prog.context([1, 2], [3, 4], copies=3) as (p, q):
+            ops.Sgate(0.7, 0) | q[1]
+            ops.BSgate(p[0]) | (q[0], q[1])
```

Comment:
Similar thought on having `Rgate` here

Reply:
Thanks Antal. This made me discover a small bug in the code. :tada: